src/rdkafka_telemetry_encode.c

```diff
@@ -183,105 +259,180 @@ void *rd_kafka_telemetry_encode_metrics(rd_kafka_t *rk, size_t *size) {
+                    (opentelemetry_proto_metrics_v1_NumberDataPoint *)rd_malloc(
+                        sizeof(opentelemetry_proto_metrics_v1_NumberDataPoint));
+                memset(data_points[i], 0,
+                       sizeof(opentelemetry_proto_metrics_v1_NumberDataPoint));
+
+                if (METRICS[metrics_to_encode[i]].is_int) {
+                        data_points[i]->which_value =
+                            opentelemetry_proto_metrics_v1_NumberDataPoint_as_int_tag;
+                        data_points[i]->value.as_int =
+                            METRICS[metrics_to_encode[i]]
+                                .calculate_value(rk)
+                                .intValue;
+
+                } else {
+                        data_points[i]->which_value =
+                            opentelemetry_proto_metrics_v1_NumberDataPoint_as_double_tag;
+                        data_points[i]->value.as_double =
+                            METRICS[metrics_to_encode[i]]
+                                .calculate_value(rk)
+                                .doubleValue;
+                }
+
+                data_points[i]->time_unix_nano = rd_clock();
+
+                //    TODO: Add data point attributes as needed
+                //    opentelemetry_proto_common_v1_KeyValue attribute =
+                //    opentelemetry_proto_common_v1_KeyValue_init_zero;
+                //    attribute.key.funcs.encode = &encode_string;
+                //    attribute.key.arg = "type";
+                //    attribute.has_value = true;
+                //    attribute.value.which_value =
+                //    opentelemetry_proto_common_v1_AnyValue_string_value_tag;
+                //    attribute.value.value.string_value.funcs.encode =
+                //    &encode_string; attribute.value.value.string_value.arg =
+                //    "heap";
+                //
+                //    data_point.attributes.funcs.encode = &encode_key_value;
+                //    data_point.attributes.arg = &attribute;
+
+
+                switch (METRICS[metrics_to_encode[i]].type) {
+
+                case METRIC_TYPE_SUM: {
+                        sum = rd_malloc(
+                            sizeof(opentelemetry_proto_metrics_v1_Sum));
+                        memset(sum, 0,
+                               sizeof(opentelemetry_proto_metrics_v1_Sum));
+                        sum->data_points.funcs.encode =
+                            &encode_number_data_point;
+                        sum->data_points.arg = data_points[i];
+                        sum->aggregation_temporality =
+                            opentelemetry_proto_metrics_v1_AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE;
+                        sum->is_monotonic = true;
+                        metrics[i]->which_data =
+                            opentelemetry_proto_metrics_v1_Metric_sum_tag;
+                        metrics[i]->data.sum = *sum;
```

Comment:
Why do we allocate a new `sum` rather than changing this directly?
Like `metrics[i]->data.sum.data_points.funcs.encode = ...` and so on?
Same for gauge

Reply:
Good catch. Updated for both